### PR TITLE
Fix race condition in Ferdi Lock at startup

### PR DESCRIPTION
**Pull request by kytwb**
### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code follows the code style of this project (run `$ yarn lint`).

### CODE REVIEW

**Comment by kytwb** [comment]
This is still a draft pull request; I need to figure out where to trigger the lock at startup properly (right after the settings are properly loaded). Removing the hack that updated `locked` multiple times did not affect the feature.

File: src/lib/Menu.js [hunk 1]
```diff
@@ -967,7 +958,7 @@ export default class FranzMenu {
 
     this.currentTemplate = tpl;
     const menu = Menu.buildFromTemplate(tpl);
-    const lockedMenu = Menu.buildFromTemplate([]);
+    const lockedMenu = new Menu();
```
Comment (vantezzen) [inline src/lib/Menu.js]: Don't we want to add some basic command to the locked menu (copy, paste, all)?

Comment (kytwb) [reply src/lib/Menu.js]: Yes, you are right. I was just focusing on the race condition with this pull request. I'll address the Lock menu in a subsequent pull request, didn't want to add noise.

**Comment by vantezzen** [comment]
I can't test the feature today but the code looks ok.

File: src/lib/Menu.js (hunk 1 above)
Comment (kytwb) [inline src/lib/Menu.js]: @vantezzen my bad about this one, I didn't mean to push it; I wanted to see if `new Menu()` would come with a default menu template or not. I can revert if needed to merge; I will try to come up with a pull request to fix `lockedMenu` properly asap.

**Comment by kytwb** [comment]
@vantezzen No worries 🙏 Please test it when you can and we'll merge it then, no hurries. A double check wouldn't hurt as we're lacking a test suite and regressions have been/can be introduced easily.

**Comment by vantezzen** [comment]
Feature is broken as I already suspected in my comment above:
The password lock only activates when there is a _change_ of the `locked` setting. As you removed the code thats sets it to `false` first, it doesn't activate anymore in this situation:
1. Password Lock gets activated
2. Ferdi gets reloaded (or restarted)
=> Password Lock isn't activated and clicking the Lock icon doesn't do anything (see video below)

PR393Demo

**Comment by kytwb** [comment]
Ah, I understand now! Coming up with a fix for this case. I removed the setTimeout as if something work with a setTimeout and doesn't without, it's usually sign of an underlying issue/race condition. Let me know what you think of the solution proposed via 984d1f6 🙏 Thank you!

**Comment by vantezzen** [comment]
Looks good, will test it later. Can you add the same logic to the lock button and lock option in the menubar? Just to make sure they don't get unresponsive once locked is somehow set to true.

**Comment by kytwb** [comment]
@vantezzen Maybe we could address the issue one level higher and make sure we always redirect to the lock screen when trying to access other screens while the application state is locked?

**Comment by vantezzen** [comment]
Thinking a little more about this, I can't think of a situation, in which these buttons could get unresponsive.
But still we could add a new `lock` action to SettingsStore that will put that locking logic into one function only instead of having multiple functions doing this same job.

**Comment by kytwb** [comment]
I am for shipping this fix before further refactoring if you're ok with it.

**Comment by kytwb** [comment]
I am for shipping this fix before further refactoring if you're ok with it.

**Comment by kytwb** [comment]
@vantezzen would you suggest I squash into one commit before merging as well next time for pull requests like this?

**Comment by vantezzen** [comment]
No, I think we should merge these PRs without squashing: It might be useful later to have the commit history instead of it being squashed - its just those CrowdIn PRs that have a lot commits that don't add any useful information to the commit history.
WDYT?

**Comment by kytwb** [comment]
That works 👌

**Pull request merged**